draft-ietf-gnap-resource-servers.md

```diff
@@ -0,0 +1,505 @@
+
+- Discovery
+- Introspection
+- Token chaining
+- Resource reference registration
+
+## RS-facing AS Discovery {#discovery}
+
+A GNAP AS offering RS-facing services can publish its features on
+a well-known discovery document using the URL `.well-known/gnap-as-rs`.
+
+This endpoint contains a JSON document consisting of a single JSON
+object with any combination of the following optional fields:
+
+introspection_endpoint:
+: The URL of the endpoint offering introspection. {{introspection}}
+
+token_formats_supported:
+: A list of token formats supported by this AS.
+
+resource_registration_endpoint:
+: The URL of the endpoint offering resource registration. {{rs-register-resource-handle}}
+
+grant_endpoint:
+: The grant endpoint of the GNAP AS.
+
+## Protecting RS requests to the AS {#authentication}
+
+Unless otherwise specified, the RS protects its calls to the AS using any of the signature
+methods defined by GNAP. This signing method MUST cover all of the appropriate
+portions of the HTTP request message, including any body elements, tokens, or
+headers required for functionality.
+
+The AS MAY require an RS to pre-register its keys
+or could alternatively allow calls from arbitrary keys, in a trust-on-first-use
+model. The RS MAY present its keys by reference or by value in
+the same fashion as a client instance calling the AS in the core protocol
+of GNAP {{I-D.ietf-gnap-core-protocol}}.
+
+## Token Introspection {#introspection}
+
+The AS issues access tokens representing a set of delegated access rights
+to be used at one or more RSs. The AS can offer an introspection service
+to allow an RS to validate that a given access token:
+
+ - has been issued by the AS
+ - has not expired
+ - has not been revoked
+ - is appropriate for the RS identified in the call
+
+When the RS receives an access token, it can call the introspection
+endpoint at the AS to get token information. 
+\[\[ [See issue #115](https://github.com/ietf-wg-gnap/gnap-core-protocol/issues/115) \]\]
+
+~~~
++--------+       +------+       +------+
+| Client |--(1)->|  RS  |       |  AS  |
+|Instance|       |      |--(2)->|      |
+|        |       |      |       |      |
+|        |       |      |<-(3)--|      |
+|        |       |      |       +------+
+|        |<-(4)--|      |               
++--------+       +------+               
+~~~
+
+1. The client instance calls the RS with its access token.
+
+2. The RS introspects the access token value at the AS.
+    The RS signs the request with its own key (not the client instance's
+    key or the token's key).
+
+3. The AS validates the access token value and the client instance's request
+    and returns the introspection response for the token.
+
+4. The RS fulfills the request from the client instance.
+
+The RS signs the request with its own key and sends the access
+token as the body of the request.
+
+~~~
+POST /introspect HTTP/1.1
+Host: server.example.com
+Content-Type: application/json
+Detached-JWS: ejy0...
```

Comment:
`resource_server` identifier could be a part of `Detached-JWS`. In this case, you might want to omit it from the actual request. I don't see any reason to specify it twice.

Reply:
The identifier is not defined as part of that structure, it's defined as part of the body of the request where needed. There's no need to overload it. DPoP overloads the key identifier here because it has no other choice.